internal/provider/resource_gitlab_runner.go

```diff
@@ -0,0 +1,257 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_runner", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_runner`" + ` resource allows registering a runner, either at an instance level
+or at a group level. The runner will be registered at a group level if the token used is from a group, or at an 
+instance level if the token used is for the instance.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/runners.html#register-a-new-runner)`,
+
+		// Since this resource updates an in-place resource, the update method is the same as the create method
+		CreateContext: resourceGitLabRunnerCreate,
+		UpdateContext: resourceGitLabRunnerUpdate,
+		ReadContext:   resourceGitLabRunnerRead,
+		DeleteContext: resourceGitLabRunnerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"token": {
+				Description: `The registration token used to register the runner`,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Sensitive:   true,
+			},
+			// Even though the output variable is just called 'token', we need to differentiate between the regitration
+			// and authentication token in terraform.
+			"authentication_token": {
+				Description: `The authentication token used for building a config.toml file. This value is not present when imported.`,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"status": {
+				Description: `The status of runners to show, one of: online and offline. active and paused are also possible values
+				 which were deprecated in GitLab 14.8 and will be removed in GitLab 16.0`,
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Description: `The runner's description'`,
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"paused": {
+				Description: `Whether the runner should ignore new jobs`,
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     gitlab.Bool(false),
```

Comment:
hmmm, what's the point of `gitlab.Bool` here? I don't think that it changes anything in behavior.
Also, we usually don't explicitly set the values to the defaults - thus, you can omit specifying the `Default` at all here ...

Reply:
@timofurrer  - Without the defaults set in the schema, the import step of the tests fail because it marks those values as "changed" (from "" -> "false")

Reply:
I've never seen this happening before 🤔 ,but okay :) The `gitlab.Bool` isn't necessary though, right?

Reply:
aah just noticed, it's because it's not being marked as `Computed` !

Reply:
@PatrickRice-KSC this is still open. WDYT about trying to mark it as `Computed` ?

Reply:
@timofurrer  - whoops, sorry! Was hidden in a compressed conversation.  Yeah, I'm good with marking it as computed! I'll add that shortly so we can get this merged 🙂